File: src/dataset/cpp/food-chain_exercism-toemm/claude/ClaudeSonnet4_food_chain_v4.cpp
#include "food_chain.h"

#include <utility>
#include <array>
#include <string>

namespace food_chain {

struct AnimalData {
    const char* name;
    const char* intro;
    const char* reaction;
};

constexpr std::array<AnimalData, 8> animals{{
    {"fly", "I know an old lady who swallowed a fly.\n", "I don't know why she swallowed the fly. Perhaps she'll die.\n"},
    {"spider", "I know an old lady who swallowed a spider.\n", "It wriggled and jiggled and tickled inside her.\n"},
    {"bird", "I know an old lady who swallowed a bird.\n", "How absurd to swallow a bird!\n"},
    {"cat", "I know an old lady who swallowed a cat.\n", "Imagine that, to swallow a cat!\n"},
    {"dog", "I know an old lady who swallowed a dog.\n", "What a hog, to swallow a dog!\n"},
    {"goat", "I know an old lady who swallowed a goat.\n", "Just opened her throat and swallowed a goat!\n"},
    {"cow", "I know an old lady who swallowed a cow.\n", "I don't know how she swallowed a cow!\n"},
    {"horse", "I know an old lady who swallowed a horse.\n", "She's dead, of course!\n"}
}};

constexpr const char* alt_bird = " that wriggled and jiggled and tickled inside her.\n";

std::string verse(int num) {
    const int idx = num - 1;
    
    if (idx == 7) {
        return std::string(animals[idx].intro) + animals[idx].reaction;
    }
    
    std::string result;
    result.reserve(256);
    
    result += animals[idx].intro;
    result += animals[idx].reaction;
    
    for (int j = idx; j > 0; --j) {
        result += "She swallowed the ";
        result += animals[j].name;
        result += " to catch the ";
        result += animals[j-1].name;
        result += (j == 2) ? alt_bird : ".\n";
        
        if (j == 1) {
            result += animals[0].reaction;
        }
    }
    
    return result;
}

std::string verses(int from, int to) {
    std::string result;
    result.reserve((to - from + 1) * 256);
    
    for (int i = from; i <= to; ++i) {
        result += verse(i);
        result += '\n';
    }
    
    return result;
}

std::string sing() {
    return verses(1, 8);
}

}  // namespace food_chain

File: src/dataset/python/food-chain_exercism-fruit-in/claude/ClaudeSonnet4_food_chain_v4.py
animals = ("fly", "spider", "bird", "cat", "dog", "goat", "cow", "horse")
phrases = (
    "It wriggled and jiggled and tickled inside her.",
    "How absurd to swallow a bird!",
    "Imagine that, to swallow a cat!",
    "What a hog, to swallow a dog!",
    "Just opened her throat and swallowed a goat!",
    "I don't know how she swallowed a cow!",
    "She's dead, of course!",
)

SPIDER_SUFFIX = " that wriggled and jiggled and tickled inside her."
FLY_ENDING = "I don't know why she swallowed the fly. Perhaps she'll die."


def recite(start_verse, end_verse):
    result = []
    for n in range(start_verse, end_verse + 1):
        result.extend(verse(n))
        if n < end_verse:
            result.append("")
    return result


def verse(n):
    animal = animals[n - 1]
    result = [f"I know an old lady who swallowed a {animal}."]
    
    if n > 1:
        result.append(phrases[n - 2])
    
    if n < 8:
        for i in range(n - 1, 0, -1):
            current_animal = animals[i]
            previous_animal = animals[i - 1]
            line = f"She swallowed the {current_animal} to catch the {previous_animal}."
            
            if previous_animal == "spider":
                line = f"She swallowed the {current_animal} to catch the spider{SPIDER_SUFFIX}"
            
            result.append(line)
        
        result.append(FLY_ENDING)
    
    return result